### PR TITLE
Cdist upsample

### PR DESCRIPTION
**Pull request by YuchengT**
Added maximize as an option in linear sum assignment.

### CODE REVIEW

File: torch_xla/csrc/ops/ops.cpp [hunk 1]
```diff
@@ -930,11 +934,11 @@ torch::lazy::NodePtr LinearSumAssignmentNode(const torch::lazy::Value& input,
         dim_shape,
         dim_shape,
     });
-    std::string shape_proto;
-    input_shape.ToProto().SerializeToString(&shape_proto);
+    std::string LSAstr = std::string(absl::bit_cast<const char*>(&LSA), sizeof(LSA));
+    //input_shape.ToProto().SerializeToString(&shape_proto);
```
Comment (ymwangg) [inline torch_xla/csrc/ops/ops.cpp]: Remove this line?

File: test/test_operations.py [hunk 2]
```diff
@@ -1841,6 +1841,14 @@ def test_emb_bf16(self):
...
     emb_out = emb(index)
     assert emb_out.dtype == torch.bfloat16
+   
+  def test_linear_sum_assignment(self):
+    xla_device = xm.xla_device()
+    xla_cost = torch.randn(20, 20, device=xla_device)
+    xla_indices = torch_xla._XLAC._xla_linear_sum_assignment(xla_cost, True)
+    cost = xla_cost.cpu()
+    indices = scipy.optimize.linear_sum_assignment(cost, True)
+    self.assertEqual(xla_indices, indices)
```
Comment (ymwangg) [inline test/test_operations.py]: Can we also test `maximize=False`?

File: third_party/xla_client/custom_cuda_ops/bernoulli.cc [hunk 3]
```diff
@@ -227,10 +228,13 @@ void LinearSumAssignment(CUstream stream, void** buffers, const char* opaque,
   int64_t* col_idx = reinterpret_cast<int64_t*>(buffers[2]);
   // std::cout << input << "," << (void*)mask << "," << output << std::endl;
 
-  xla::ShapeProto shape;
-  if (!shape.ParseFromArray(opaque, opaque_len)) {
+  if (opaque_len != sizeof(xla::LSAInfo)) {
     return;
```
Comment (ymwangg) [inline third_party/xla_client/custom_cuda_ops/bernoulli.cc:232]: Ideally we want to throw errors in `XlaCustomCallStatus`, but this is fine for now.

File: torch_xla/csrc/ops/ops.cpp (hunk 1 above)
Comment (YuchengT) [reply torch_xla/csrc/ops/ops.cpp]: Ah I see. Thanks for pointing out.

File: test/test_operations.py (hunk 2 above)
Comment (YuchengT) [reply test/test_operations.py]: Yes let me add the case.

**Comment by ymwangg** [comment]
@YuchengT Thanks!

**Pull request merged**